### PR TITLE
Mail - Reply-To can be the same as the To address (spam indicator - negative points)

### PR DESCRIPTION
**Pull request by ShaiMagal**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Check below.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Go to module "ps_emailalerts" settings, enable "new order" confirmation and fill same email address like for whole shop (main eshop email). Make any order and check your email. There shouldn't be "Reply to".
| UI Tests          |
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | https://www.openservis.cz/

### Description:
For example: if we use module ps_emailalerts and enable option informing me about New order and set up same email adress like e-shop address, we got "Reply to" same like "To".
It's spam indicator. I know, it's "valid use case", but some customers have problems, because they got for this emails any negative spam points.
Google for "REPLYTO_EQ_TO_ADDR".
(This can be related to more modules, not only ps_emailalerts, so we need to deal with it with file Mail.php)

### Screenshots:
Default rule from Rspamd:
A
Example of same Reply to:
B
Real impact:
C

**After apply this PR(commit) problem disappeared**

### CODE REVIEW

**Comment by Hlavtox** [comment]
I will send you our config, but we don't have any issues with this. Do you have some source where it is marked as spam, so I can check it out?

**Comment by kpodemski** [comment]
I've never seen any problems with this approach 🤔

**Comment by Hlavtox** [comment]
This is it?

https://github.com/rspamd/rspamd/blob/d94a2b8ae8842ff9525107bef3f29a814aae8fdb/rules/headers_checks.lua#L332

**Review by tleon — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]
After consideration, I think it makes sense, we don't need `replyTo` if the address is the same as the sender.

but I would appreciate @Hlavtox review as well, then we can go to QA 😎

**Review by boherm — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]
Approving logic wise. Make sure it works if receiver(s) are string or array.

**Comment by ShaiMagal** [comment]
@Hlavtox 
I already post screenshots from real world spamfilter rspamd, I don't tell it fall to spam because of this, but this is one "spam indicator" and mail got negative point(s) :)
And it's useless to use reply-to, if this is same like sender, like already told @kpodemski  :) It's "duplicite", so I think, there will be only positive impact, not negative.

btw, @Hlavtox - Write "REPLYTO_EQ_TO_ADDR" to google and look for first posts, there are more complains from other customers, from other platform as well (for example first one is WordPress) - not much informations on Google, because it's logical. Name "REPLYTO_EQ_TO_ADDR" is internal, so not a lot of users will know this :)


If someone can test it as well, it can be great to be sure, I didn't miss something! :)

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
